dune

```diff
@@ -114,17 +114,15 @@
  (libraries stdcompat))
 
 (rule
- (target stdcompat__native.ml)
- (deps stdcompat__native.ml_native)
- (action (copy %{deps} %{target})))
+ (action (copy stdcompat__native.ml_native stdcompat__native.ml)))
```

Comment:
Are you this is correct? If you don't declare an explicit target, is dune able to calculate the built dependency tree correctly, or will this only be created when you run `build @all`?

Reply:
If I understand [inferred rules](https://dune.readthedocs.io/en/stable/dune-files.html#inferred-rules] correctly, Dune is able to infer that there is such a dependency and it generates such a target, so when other rules require `stdcompat__native.ml` it should be able to determine that this rule needs to run first.

But in fact according to the example, I can also leave out the `action`, to make it even shorter. The build seems to work :slightly_smiling_face: